### PR TITLE
Add Sui support

### LINKED ISSUE #889: Add Sui support

### PR DESCRIPTION
**Pull request by yvebe**
Fix #889

### CODE REVIEW

**Review by johnnyluo — APPROVED** [review]
LGTM

File: data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        val inputData = getPreSignedInputData(keysignPayload)
+        Timber.d("input data: ${inputData.toHexString()}")
+
+        val hashes = TransactionCompiler.preImageHashes(coinType, inputData)
+        val preSigningOutput =
+            wallet.core.jni.proto.TransactionCompiler.PreSigningOutput.parseFrom(hashes)
+
+        require(preSigningOutput.errorMessage.isEmpty()) { preSigningOutput.errorMessage }
+
+        return listOf(preSigningOutput.dataHash.toByteArray().toHexString())
```
Comment (enriquesouza) [inline data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt:66]: Convert to black2b hash

File: data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        signatures: Map<String, KeysignResponse>
+    ): SignedTransactionResult {
+        val pubkeyData = vaultHexPubKey.hexToByteArray()
+        val publicKey = PublicKey(pubkeyData, PublicKeyType.ED25519)
+
+        val inputData = getPreSignedInputData(keysignPayload)
+        val hashes = TransactionCompiler.preImageHashes(coinType, inputData)
+        val preSigningOutput = wallet.core.jni.proto.TransactionCompiler.PreSigningOutput
+            .parseFrom(hashes)
+        val preSigningOutputDataBlake2b =preSigningOutput.dataHash.toByteArray()
```
Comment (enriquesouza) [inline data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt:81]: Convert to black2b hash

File: data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt (hunk 1 above)
Comment (enriquesouza) [reply data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt:66]: Example ```const blakeHash = this.walletCore.Hash.blake2b(preSigningOutput.data, 32);```

File: data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt (hunk 2 above)
Comment (enriquesouza) [reply data/src/main/kotlin/com/vultisig/wallet/data/crypto/SuiHelper.kt:81]: const blakeHash = this.walletCore.Hash.blake2b(preSigningOutput.data, 32);

**Review by enriquesouza — COMMENTED** [review]
If you have issues signing, broadcasting and confirming on the blockchain you should apply those 

const blakeHash = this.walletCore.Hash.blake2b(preSigningOutput.data, 32);

**Pull request merged**